### PR TITLE
Vertical grid lines using DOM elements

### PR DESCRIPTION
**Pull request by rbrtsmith**
* Addresses #59 
* ~~No tests yet~~
* Comparing with the approach on #60

### CODE REVIEW

File: src/components/Timeline/Grid/index.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+const VerticalGrid = ({ time, cells }) =>
```
Comment (WebSeed) [inline src/components/Timeline/Grid/index.jsx]: Guess it should be `Grid`

**Comment by WebSeed** [comment]
Looking good @rbrtsmith

**Comment by WebSeed** [comment]
As to which `timebar` is used something like:

```js
[
  { title: 'Row 1' },
  { title: 'Row 2', useAsGrid: true }
]
```

**Comment by rbrtsmith** [comment]
> As to which timebar is used something like:
>  ```[
>    { title: 'Row 1' },
>    { title: 'Row 2', useAsGrid: true }
>  ]
>  ```

@WebSeed  Should we make `useAsGrid` be a required field on at least one of rows or use a default row?

**Comment by WebSeed** [comment]
Got to be optional and if multiple `useAsGrid` provided we take the first only (for now)

File: src/utils/getGridRowCells.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const getGridRowCells = timebar =>
```
Comment (WebSeed) [inline src/utils/getGridRowCells.js]: This function could just be something like:

```js
const getGrid = (timebar.find(row => row.useAsGrid) || {}).cells
```

File: src/utils/getGridRowCells.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  timebar.find(row => (
+    row.useAsGrid) ||
+    timebar[timebar.length - 1]
```
Comment (WebSeed) [inline src/utils/getGridRowCells.js]: Don't think we want a default - this is a feature users should be able to opt into.

File: src/scss/components/_grid.scss [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+.rt-grid {
+  left: 0;
+  right: 0;
+}
+
+.rt-grid__cell {
+  background: #fff;
+  box-shadow: 1px 0 0 0 $react-timelines-separator-color inset;
```
Comment (WebSeed) [inline src/scss/components/_grid.scss]: Can we rename this var to something like `$react-timelines-keyline-color`?

File: src/components/Timeline/Grid/index.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import getGridRowCells from '../../../utils/getGridRowCells'
+
+const Grid = ({ time, timebar }) =>
+  <div className="rt-grid">
+    {getGridRowCells(timebar).map(({ id, start, end }) =>
```
Comment (WebSeed) [inline src/components/Timeline/Grid/index.jsx]: This component shouldn't be provided with the whole `timebar` (it shouldn't even know what a `timebar` is!). Instead `getGridCells` should be called a few layers higher. Sometimes a bit of a smell when you pass a load of data to a function for it to then have to filter out and find the bits it needs (when the filtering could have been done at a higher level). I.e. would expect 

```jsx
<Body time={time} grid={grid} tracks={tracks} clickElement={clickElement} />
```

In `src/components/Timeline/index.jsx`

File: src/components/Timeline/index.jsx [hunk 6]
```diff
@@ -50,6 +52,7 @@ class Timeline extends Component {
       pointerVisible,
       pointerHighlighted
     } = this.state
+    const grid = getGridCells(timebar)
```
Comment (WebSeed) [inline src/components/Timeline/index.jsx]: I guess this should be called `buildGrid`

File: src/utils/getGridCells.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
+const getGridCells = timebar =>
```
Comment (WebSeed) [inline src/utils/getGridCells.js]: `findGrid`

**Pull request merged**